pkg/filesystem/filesystem_test.go

```diff
@@ -149,6 +87,95 @@ func TestFileSystems(t *testing.T) {
 	}
 }
 
+func loadFS(fileSys filesystem.Filesystem) ([]FileInfo, error) {
+	var err error
+	var files []FileInfo
+
+	permMask := fs.FileMode(0111)
+	if runtime.GOOS == "windows" {
+		permMask = 0
+	}
+
+	err = fs.WalkDir(fileSys, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		fi, err := fileSys.Stat(path)
+		if err != nil {
+			return err
+		}
+		var bs []byte
+		switch fi.Mode() & fs.ModeType {
+		case 0:
+			f, err := fileSys.Open(path)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			bs, err = io.ReadAll(f)
+			if err != nil {
+				return err
+			}
+		case fs.ModeSymlink:
+			t, _ := fileSys.Readlink(path)
+			bs = []byte(t)
+		}
+		files = append(files, FileInfo{
+			Path:       path,
+			Typ:        fi.Mode().Type(),
+			Executable: fi.Mode()&permMask == permMask && !fi.IsDir(),
+			Content:    bs,
+		})
+		return nil
+	})
+
+	return files, err
+}
+
+func loadLocalFiles(root string) ([]FileInfo, error) {
+	var files []FileInfo
+	var err error
+
+	permMask := fs.FileMode(0111)
+	if runtime.GOOS == "windows" {
+		permMask = 0
+	}
+
+	err = filepath.Walk(root, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		fi, err := os.Lstat(path)
+		if err != nil {
+			return err
+		}
+		var bs []byte
+		switch fi.Mode() & fs.ModeType {
+		case 0:
+			bs, err = os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+		case fs.ModeSymlink:
+			t, _ := os.Readlink(path)
+			bs = []byte(t)
+		}
+		path, err = filepath.Rel(root, path)
+		if err != nil {
+			return err
+		}
+		files = append(files, FileInfo{
+			Path:       filepath.ToSlash(path),
```

Comment:
How does it work with Windows backslash path separator?

Reply:
The Go's `fs` package uses slashes (like zip or tar does).

Reply:
This line is here precisely so it works on Windows.